File: src/my_project_typescript/easy_problems_typescript/from101to150/largestPerimeterTriangle.ts
function largestPerimeter(nums: number[]): number {
    nums =  nums.sort((a,b) => a>b?-1:1)
    let lenNums: number = nums.length
    let answer:  number = 0

    for (let i = 2; i < lenNums; i++){
        if (nums[i-2] - nums[i-1] < nums[i]){
            answer = nums[i] + nums[i-1] + nums[i-2]
            return answer 
        }
    }
    return answer 
};


console.log(largestPerimeter([2,1,2]))

console.log(largestPerimeter([1,2,1]))

console.log(largestPerimeter([3,2,3,4]))

console.log(largestPerimeter([3,6,2,3]))

File: src/my_project/easy_problems/from201to250/largest_perimeter_triangle.py
from typing import List 

class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        
        nums.sort(reverse=True)
        len_nums: int = len(nums)
        answer: int = 0
        
        for i in range(2,len_nums):
            if nums[i-2] - nums[i-1] <nums[i]:
                answer = nums[i] + nums[i-1] + nums[i-2]
                return answer 
            
        
        return answer 
    
    
solution = Solution()

print(solution.largestPerimeter(nums = [2,1,2]))

print(solution.largestPerimeter(nums = [1,2,1]))

print(solution.largestPerimeter(nums = [3,2,3,4]))

print(solution.largestPerimeter(nums = [3,6,2,3]))
